src/components/Product.js

```diff
@@ -1,9 +1,51 @@
-import React from 'react'
+import React, { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { useDispatch } from 'react-redux';
+import { addToCart } from "../redux/coderSlice";
 
 const Product = () => {
-    return (
-        <div>Product</div>
-    )
-}
+  const dispatch = useDispatch();
+  const [productPageInfo, setProductPageInfo] = useState({});
+  const [baseQuant, setBaseQuant] = useState(1);
+  const location = useLocation();
+  useEffect(() => {
+    setProductPageInfo(location.state.item);
+  }, []);
 
-export default Product
\ No newline at end of file
+  const productName = location.state.item.name.stringValue;
+  const productImage = location.state.item.image.stringValue;
+  const productDescription = location.state.item.description.stringValue;
+  const productPrice = location.state.item.price.integerValue;
+
+  //   console.log(productPageInfo.name.stringValue);
+
+  return (
+    <div>
+      <div>Product Info page</div>
+      <h2>{productName}</h2>
+      <div>
+        <img style={{ width: "150px" }} src={productImage} alt="" />
+        <p>{productDescription}</p>
+        <p>${productPrice}</p>
+      </div>
+      <div>
+        <button onClick={() => setBaseQuant(baseQuant === 1 ? baseQuant = 1 : baseQuant - 1)}>decrease quantity</button>
```

Comment:
This code looks a bit weird. 
1) `baseQuant = 1` :  I think it should just be 1.
2) Overall, seems like there is no need to set `baseQuant` to 1 because you are already checking if it's 1.